### ISSUE TITLE
sql: ANY comparisons should return false with empty RHS

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#98216]
`ANY` comparisons should return `false` when the RHS is empty, regardless of the LHS of the comparison. Here's an example in Postgres:

```
marcus=# SELECT NULL = ANY(ARRAY[]::INT[]);
 ?column?
----------
 f
(1 row)
```

We incorrectly return `NULL` in this case in v22.2.2:

```
defaultdb> select NULL = ANY(ARRAY[]::INT[]);
  ?column?
------------
  NULL
(1 row)
```

I suspect that a normalization rule is at fault here. In addition, it looks like the legacy normalization code here is incorrect, too: https://github.com/cockroachdb/cockroach/blob/fc1fa24f38a0b4f1e60bd079f231de37a2e842bd/pkg/sql/sem/normalize/normalize_exprs.go#L289-L292

Jira issue: CRDB-25136

**Comment by mgartner** [comment]
Here's another example.

In Postgres:

```
marcus=# CREATE TABLE empty (a INT);
CREATE TABLE

marcus=# SELECT NULL = ANY (SELECT a FROM empty);
 ?column?
----------
 f
(1 row)
```

In CRDB:

```
defaultdb> CREATE TABLE empty (a INT);
CREATE TABLE

defaultdb> SELECT NULL = ANY (SELECT a FROM empty);
  ?column?
------------
  NULL
(1 row)
```

When the expression is not folded at optimization time, we get the correct result.

In Postgres:

```
marcus=# SELECT i, i = ANY (SELECT a FROM empty) FROM (VALUES (1), (NULL)) v(i);
  i   | ?column?
------+----------
    1 | f
 NULL | f
(2 rows)
```

In CRDB:

```
defaultdb> SELECT i, i = ANY (SELECT a FROM empty) FROM (VALUES (1), (NULL)) v(i);
   i   | ?column?
-------+-----------
     1 |    f
  NULL |    f
(2 rows)
```

**Comment by rafiss** [comment]
this is similar to https://github.com/cockroachdb/cockroach/issues/37793, though i'm not sure if the resolution to that issue covered subqueries

**Comment by mgartner** [comment]
All the cases above were fixed by #97948. Closing.
